lib/http/2/connection.rb

```diff
@@ -226,6 +226,12 @@ def receive(data)
         else
           case frame[:type]
           when :headers
+            # When even-numbered stream identifier is received,
+            # the endpoint MUST respond with a connection error of type PROTOCOL_ERROR.
+            if frame[ :stream ].even?
```

Comment:
Err, wouldn't this break client apps? As in, connection.rb is used by both client and server classes, and this check effectively forces client-only code path for header processing.